pkg/schema/tracker.go

```diff
@@ -329,3 +368,227 @@ func (tr *Tracker) CreateTableIfNotExists(table *filter.Table, ti *model.TableIn
 func (tr *Tracker) GetSystemVar(name string) (string, bool) {
 	return tr.se.GetSessionVars().GetSystemVar(name)
 }
+
+// GetDownStreamIndexInfo gets downstream PK/UK(not null) Index.
+// note. this function will init downstreamTrack's table info.
+func (tr *Tracker) GetDownStreamIndexInfo(tctx *tcontext.Context, tableID string, originTi *model.TableInfo, downstreamConn *conn.BaseConn) (*model.IndexInfo, error) {
+	dti, ok := tr.downstreamTracker.tableInfos[tableID]
+	if !ok {
+		log.L().Info("DownStream schema tracker init. ", zap.String("tableID", tableID))
+		ti, err := tr.getTiByCreateStmt(tctx, tableID, downstreamConn)
+		if err != nil {
+			return nil, err
+		}
+		dti = getDownStreamTi(ti, originTi)
+		tr.downstreamTracker.tableInfos[tableID] = dti
```

Comment:
we should use `lock` or `sync.Map` to make the access of `tr.downstreamTracker.tableInfos` is thread safe, becasue there are many goroutines use this func at same time

Reply:
ok, i will add `lock` or `sync.Map`.